app/app.tsx

```diff
@@ -87,6 +91,58 @@ const CustomToast = () => {
   return <Toast config={toastConfig} topOffset={insets.top} />
 }
 
+// Setting up our OTA Updates component
+const OTAUpdates = () => {
+  const [isModalVisible, setIsModalVisible] = useState(false)
+
+  async function fetchAndRestartApp() {
+    const fetchUpdate = await Updates.fetchUpdateAsync()
+    if (fetchUpdate.isNew) {
+      await Updates.reloadAsync()
+    } else {
+      setIsModalVisible(false)
+      reportCrash("Fetch Update failed")
+    }
+  }
+
+  async function onFetchUpdateAsync() {
+    if (__DEV__ || process.env.NODE_ENV === "development") return
+    try {
+      const update = await Updates.checkForUpdateAsync()
+      setIsModalVisible(update.isAvailable)
+    } catch (error) {
+      reportCrash(error)
+    }
+  }
+
+  useAppState({
+    match: /background/,
+    nextAppState: "active",
+    callback: onFetchUpdateAsync,
+  })
+
+  return (
+    <Modal
+      title="ota.title"
+      subtitle="ota.subtitle"
+      confirmOnPress={{
+        cta: () => {
+          fetchAndRestartApp()
```

Comment:
do we need to await this? or just put a label somewhere with some progess like "updating..." ?

Reply:
updated in next push: https://github.com/infinitered/ChainReactApp2023/pull/183/commits/2f3ab73f3a3ebd9cf43e62516ec90012aad73f0c